Document build steps using just esy (#177)
melange-re/melange: CONTRIBUTING.md

- The [Nix](https://nixos.org/) package manager

## Instalation

> You can both use [#nix](Nix) or [#esy](Esy) to setup your application, give preference to
> Nix since it can have more predictable environments, but you can run with Esy if your prefer.

Reviewer: ```suggestion
Melange can be set up with [#nix](Nix) or [#esy](Esy). Instructions for both are detailed below.
```

Once you're in the shell, you have access to `dune`, `node`, `yarn`, and all the other executables you need to get down to business.

### Esy

> You can run without using Nix, you just need [esy](https://esy.sh/). Just straightforward, run `esy` and you are set up.
> You still need to make sure you runned `npm install` on the `ocaml-tree`directory.

Reviewer: ```suggestion
In order to develop using [Esy](https://esy.sh/), you need to install it first. The usual way for doing that is to run `npm install -g esy`. Then run:

1. `esy install` to get all the project's dependencies
2. `esy build` to build the project.
```

## Developing

Before you try building with `dune`, be sure to install the local `ocaml-tree` project:

```sh
cd ocaml-tree && npm install
```

> And, here are some common commands you may find useful:

Reviewer: ```suggestion
Here are some common commands you may find useful:
```

```

If that all passes, then congratulations! You are well on your way to becoming a contributor 🎉

> ## Without NixOS

Reviewer: I suggest adding a `Nix` section about, and name this section `Esy`
